logic/src/commonMain/kotlin/com/wire/kalium/logic/sync/receiver/TeamEventReceiver.kt

```diff
@@ -68,25 +81,50 @@ internal class TeamEventReceiverImpl(
             .onSuccess {
                 val knownUser = userRepository.getKnownUser(userId).first()
                 if (knownUser?.name != null) {
-                    val conversationIds = conversationRepository.getConversationIdsByUserId(userId)
-                    conversationIds.onSuccess {
-                        it.forEach { conversationId ->
-                            val message = Message.System(
-                                id = uuid4().toString(), // We generate a random uuid for this new system message
-                                content = MessageContent.TeamMemberRemoved(knownUser.name),
-                                conversationId = conversationId,
-                                date = event.timestampIso,
-                                senderUserId = userId,
-                                status = Message.Status.SENT,
-                                visibility = Message.Visibility.VISIBLE
+                    conversationRepository.getConversationIdsByUserId(userId)
```

Comment:
@Garzas @vitorhugods  did I change the logic here by mistake? why are there 2 top level success blocks?

Reply:
You're correct.

```kotlin
either.onSuccess { A() }
    .onSuccess { B() }
    .onSuccess { C() }
```

Is the same as.
```kotlin
either.onSuccess { 
    A()
    B()
    C()
}
```

Probably some refactoring oversight made in the past?